auth-jwt-service/src/main/java/io/stargate/auth/jwt/AuthzJwtService.java

```diff
@@ -139,27 +138,22 @@ private JSONObject extractClaimsFromJWT(String token) throws JSONException {
     return payload.getJSONObject(CLAIMS_FIELD);
   }
 
-  private void preCheckDataReadWrite(
-      JSONObject stargateClaims, List<String> primaryKeyValues, Table tableMetadata)
+  private void preCheckDataReadWrite(JSONObject stargateClaims, List<TargetCell> targetCells)
       throws JSONException, UnauthorizedException {
-    List<Column> keys = tableMetadata.primaryKeyColumns();
-
-    if (primaryKeyValues.size() > keys.size()) {
-      throw new IllegalArgumentException("Provided more primary key values than exists");
-    }
-
-    for (int i = 0; i < primaryKeyValues.size(); i++) {
+    for (TargetCell targetCell : targetCells) {
       // If one of the columns exist as a field in the JWT claims and the values do not match then
       // the request is not allowed.
-      if (stargateClaims.has(STARGATE_PREFIX + keys.get(i).name())) {
-
-        if (!Column.ofTypeText(keys.get(i).type())) {
+      if (stargateClaims.has(STARGATE_PREFIX + targetCell.getName())) {
+        String targetCellType = targetCell.getType().toLowerCase();
+        if (!(targetCellType.equals(Type.Varchar.cqlDefinition())
+            || targetCellType.equals(Type.Text.cqlDefinition()))) {
```

Comment:
If we processed whole statements in `authorizedDataRead`  (extracting where clauses) it should be possible to use type enum values for comparison.